metrics/web/docs/features/deep_links/01_deep_linking.md

```diff
@@ -399,3 +412,153 @@ class MetricsRouteInformationParser {
+```dart
+/// Returns the [PageParametersModel] based on the given [configuration].
+PageParametersModel create(RouteConfiguration configuration) {
+  final parameters = configuration.parameters;
+  
+  switch(configuration.name) {
+    case RouteName.dashboard:
+      return DashboardPageParameters.fromMap(parameters);
+    case RouteName.projectGroups:
+      // Other page parameters handling
+  }
+}
+```
+
+The `PageParametersFactory` should be injected into `NavigationNotifier` since the `NavigationNotifier` holds all the navigation-related logic of the application and controls its deep linking. Consider the [`NavigationNotifier`](#navigationnotifier) section to know more details in notifier-related changes.
+
+##### MetricsPage and MetricsPageFactory
+To make the application able to restore the `PageParametersModel` in response to navigation events (e.g., popping a page), we should update the `MetricsPage` and `MetricsPageFactory` classes.
+
+A `MetricsPage` should store a `RouteName` that was used to create this page and the `PageParametersModel` as an `arguments` field. The `MetricsPage` class should also provide an overridden `copyWith` method to copy a page with the updated parameters.
+
+We should also update the `MetricsPageFactory` to create `MetricsPage`s with the given `RouteName` and `PageParametersModel`.
+
+The code snippet below demonstrates the `MetricsPageFactory.create` changes:
+```dart
+  /// Creates the [MetricsPage] from the given [RouteConfiguration].
+  MetricsPage create({
+    RouteName routeName, 
+    PageParametersModel pageParameters,
+  }) {
+    switch (routeName) {
+      case RouteName.loading:
+        return MetricsPage(
+          child: const LoadingPage(),
+          routeName: routeName,
+          arguments: pageParameters,
+        );
+    /// Other MetricsPages creation...
+```
+
+##### NavigationNotifier
+The `NavigationNotifier` is a class that holds the navigation logic of the Metrics Web application. The following subsections describe the main changes to the`NavigationNotifier` related to the `deep linking` feature implementation.
+
+###### Updating page parameters
+When the `NavigationNotifier` receives a new `RouteConfiguration`, it should create a new `PageParametersModel` using a new `RouteConfiguration` instance and notify all listeners about the updates.
+
+To do that, we should implement a new method `_updatePageParameters()` and call it whenever the current `RouteConfiguration` changes (in the `pop()` and the `_addNewPage()` methods).
+
+The following diagram demonstrates the general concept for updating the `PageParametersModel` when the current `RouteConfiguration` changes:
+![Updating page parameters diagram](http://www.plantuml.com/plantuml/proxy?cache=no&fmt=svg&src=https://raw.githubusercontent.com/Flank/flank-dashboard/deep_links_design_improvements/metrics/web/docs/features/deep_links/diagrams/updating_page_parameters_sequence_diagram.puml)
```

Comment:
:readable:
This appears to be badly formatted. It seems like the empty line is missed: 
```suggestion
The following diagram demonstrates the general concept for updating the `PageParametersModel` when the current `RouteConfiguration` changes:

![Updating page parameters diagram](http://www.plantuml.com/plantuml/proxy?cache=no&fmt=svg&src=https://raw.githubusercontent.com/Flank/flank-dashboard/deep_links_design_improvements/metrics/web/docs/features/deep_links/diagrams/updating_page_parameters_sequence_diagram.puml)
```